DirectEmployers/MyJobs
# Use permissions for reporting.

## What's this PR do?

* Removes an old hack and replaces it with checking permissions.

## Screenshots (if appropriate)

Not changed.

## Where should the reviewer start?

* Note changes to `update_activities` which fixed a bug.
* `ReportingTypeManager` has a more sophisticated way of getting active report types.
* The unit test feature has activities assigned to reporting types.
* Some view permissions are removed. (I'm a little unsure if I'm ok removing the one on `dynamicoverview` but I think it's ok.

## How should this be manually tested?

Add and remove permissions to a user's role and see how their report intention choices change.

## Any background context you want to provide?

none.
